src/AppInstallerCLICore/Workflows/WorkflowBase.cpp

```diff
@@ -731,26 +732,50 @@ namespace AppInstaller::CLI::Workflow
                 if (updateAvailable || !m_onlyShowUpgrades)
                 {
                     Utility::LocIndString availableVersion, sourceName;
+                    Utility::LocIndString packageId = match.Package->GetProperty(PackageProperty::Id);
 
                     if (latestVersion)
                     {
+                        // Always show the source for correlated packages
+                        sourceName = latestVersion->GetProperty(PackageVersionProperty::SourceName);
+
                         if (updateAvailable)
                         {
                             availableVersion = latestVersion->GetProperty(PackageVersionProperty::Version);
-                            availableUpgradesCount++;
+                            if (context.Args.Contains(Execution::Args::Type::ListAll) || !packageIdsPrinted.count({ packageId, sourceName }))

```

Comment:
Sorry, I may not make clear in previous comment. SourceName is for display. SourceIdentifier should be used to identify a source. We should use value from PackageVersionProperty::SourceIdentifier for the duplicate check.

Reply:
Resolved in latest commit!